vm/systemSmartContracts/auction.go

```diff
@@ -865,6 +869,12 @@ func (s *stakingAuctionSC) activateStakingFor(
 			s.eei.Finish(currentBLSKey)
 			s.eei.Finish([]byte{waiting})
 		}
+		if s.flagStakingV2.IsSet() {
+			okSet := s.setOwnerOfBlsKey(currentBLSKey, ownerAddress)
```

Comment:
instead of having 2 calls you could add to in the "stake" function the owner address as well.

Reply:
Yes, and `executeOnStakingSC` being just a proxy function anyway that calls `ExecuteOnDestContext` could have the logic of assembling the correct parameters depending on the version

Reply:
Grossly violating OCP for the sake of optimizing things...will do

Reply:
done